### ISSUE TITLE
[REC] The bug with cooldown(offset) for primary priority

### ISSUE THREAD

**Issue by savelichev** [Hekili/hekili#3441]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

Cata Classic. Warlock. Demon. 
In general addon works nice.
But when i cast "Ritual of Souls" it provides 1 min cooldown(offset) for all priorities, like addon is  paused.
This offset value is a chanelling time of the  e.g. "Ritual of Souls" spell.
Even when i apply dots it shows them in prio list.
So addon does not work until this cooldown passed
![image](https://github.com/user-attachments/assets/ae67d42f-a72c-49c3-ac65-d5e5acd2b460)
When i cast "Ritual of summoning" it provides 2min cooldown
![image](https://github.com/user-attachments/assets/f9463a56-8eb2-4623-b806-6f8cc4c9dfbb)

### How to Reproduce

1. Enter the game as Warlock
2. Go to the training dummy
3. Enable addon
4. Before the battle click on the "Ritual of Souls" or "Ritual of Summoning"

### Snapshot (Link)

https://pastebin.com/9XTZTwz1

**Comment by georgedeam** [comment]
it will also do this for summoning, i fixed this in the meantime by commenting out the two ritual spells in warlock.lua

**Comment by Hekili** [comment]
It should do this while you're casting/channeling (even with the hidden cast bar), as long as it stops when you complete or interrupt your ritual.  Are you saying that it stays like that afterward?

**Comment by georgedeam** [comment]
yes it stays the entire 2 minutes of the summon cooldown

**Comment by mykolasmiro** [comment]
yes, it take like the channel time for cast 1min for ritual of souls and 2min for ritual for summoning, and apply this time as an offset for rotation even if you stop channeling. So if start cast right before the battle, rotation will be like on pause until this offset passed. The tip from @georgedeam, to comment both spells in the warlock sources helped.

**Comment by savelichev** [comment]
yes, it take like the channel time for cast 1min for ritual of souls and 2min for ritual for summoning, and apply this time as an offset for rotation even if you stop channeling. So if start cast right before the battle, rotation will be like on pause until this offset passed. The tip from @georgedeam, to comment both spells in the warlock sources helped.

**Comment by georgedeam** [comment]
It appears that it is commented out on the retail version, maybe an oversight for cata?

```
--[[ Begins a ritual that sacrifices a random participant to summon a doomguard. Requires the caster and 4 additional party members to complete the ritual.
    ritual_of_doom = {
        id = 342601,
        cast = 0,
        cooldown = 3600,
        gcd = "spell",
        school = "shadow",

        spend = 1,
        spendType = "soul_shards",

        startsCombat = false,

        toggle = "cooldowns",

        handler = function ()
        end,
    },

    -- Begins a ritual to create a summoning portal, requiring the caster and 2 allies to complete. This portal can be used to summon party and raid members.
    ritual_of_summoning = {
        id = 698,
        cast = 0,
        cooldown = 120,
        gcd = "spell",
        school = "shadow",

        spend = 0.05,
        spendType = "mana",

        startsCombat = false,

        toggle = "cooldowns",

        handler = function ()
        end,
    }, ]]
```

**Comment by Smufrik** [comment]
@Hekili this is revolved
